src/main/asciidoc/java/http.adoc

```diff
@@ -1214,6 +1524,185 @@ By default pipe-lining is disabled.
+
+[source,java]
+----
+HttpServer server = vertx.createHttpServer(http2Options);
+
+server.connectionHandler(connection -> {
+  System.out.println("A client connected");
+});
+----
+
+NOTE: this only applies to the HTTP/2 protocol
+
+==== Client connections
+
+The `link:../../apidocs/io/vertx/core/http/HttpClientRequest.html#connection--[connection]` method returns the request connection on the client:
+
+[source,java]
+----
+HttpConnection connection = request.connection();
+----
+
+A connection handler can be set on the request to be notified when the connection happens:
+
+[source,java]
+----
+request.connectionHandler(connection -> {
+  System.out.println("Connected to the server");
+});
+----
+
+NOTE: this only applies to the HTTP/2 protocol
+
+==== Connection settings
+
+The configuration of an HTTP/2 is configured by the `link:../../apidocs/io/vertx/core/http/Http2Settings.html[Http2Settings]` data object.
+
+Each endpoint must respect the settings sent by the other side of the connection.
+
+When a connection is established, the client and the server exchange initial settings. Initial settings
+are configured by `link:../../apidocs/io/vertx/core/http/HttpClientOptions.html#setInitialSettings-io.vertx.core.http.Http2Settings-[setInitialSettings]` on the client and
+`link:../../apidocs/io/vertx/core/http/HttpServerOptions.html#setInitialSettings-io.vertx.core.http.Http2Settings-[setInitialSettings]` on the server.
+
+The settings can be changed at any time after the connection is established:
+
+[source,java]
+----
+connection.updateSettings(new Http2Settings().setMaxConcurrentStreams(100));
+----
+
+As the remote side should acknowledge on reception of the settings update, it's possible to give a callback
+to be notified of the acknowledgent:
+
+[source,java]
+----
+connection.updateSettings(new Http2Settings().setMaxConcurrentStreams(100), ar -> {
+  if (ar.succeeded()) {
+    System.out.println("The settings update has been acknowledged ");
+  }
+});
+----
+
+Conversely the `link:../../apidocs/io/vertx/core/http/HttpConnection.html#remoteSettingsHandler-io.vertx.core.Handler-[remoteSettingsHandler]` is notified
+when the new remote settings are received:
+
+[source,java]
+----
+connection.remoteSettingsHandler(settings -> {
+  System.out.println("Received new settings");
+});
+----
+
+==== Connection ping
+
+HTTP/2 connection ping is useful for determining the connection round-trip time or check the connection
+validity: `link:../../apidocs/io/vertx/core/http/HttpConnection.html#ping-io.vertx.core.buffer.Buffer-io.vertx.core.Handler-[ping]` sends a `PING` frame to the remote
+endpoint:
+
+[source,java]
+----
+Buffer data = Buffer.buffer();
+for (byte i = 0;i < 8;i++) {
+  data.appendByte(i);
+}
+connection.ping(data, pong -> {
+  System.out.println("Remote side replied");
+});
+----
+
+Vert.x will send automatically an acknowledgement when a `PING` frame is received,
+an handler can be set to be called when the connection is pinged:
+
```

Comment:
yes, the ACK is actually sent by Netty HTTP/2 automatically and not by Vert.x . This is just a mean to be aware of the ping, a protocol like gRPC uses the HTTP/2 ping frames and need this callback.


> On Apr 6, 2016, at 10:10 AM, Clement Escoffier <notifications@github.com> wrote:
> 
> In src/main/asciidoc/java/http.adoc <https://github.com/eclipse/vert.x/pull/1353#discussion_r58666537>:
> 
> > +endpoint:
> > +
> > +[source,java]
> > +----
> > +Buffer data = Buffer.buffer();
> > +for (byte i = 0;i < 8;i++) {
> > +  data.appendByte(i);
> > +}
> > +connection.ping(data, pong -> {
> > +  System.out.println("Remote side replied");
> > +});
> > +----
> > +
> > +Vert.x will send automatically an acknowledgement when a `PING` frame is received,
> > +an handler can be set to be called when the connection is pinged:
> > +
> The the ACK still sent when the handler is set ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/eclipse/vert.x/pull/1353/files/c7a6a1ce8a1da75fc0edcccf686f25b82162b149#r58666537>